main.rb

```diff
@@ -0,0 +1,151 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    array.length.times { |index| yield(array[index]) }
+    array
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    array.length.times { |index| yield(array[index], index) }
+    array
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    array = []
+    my_each { |index| array.push(index) if yield(index) }
+    array
+  end
+
+  def my_all?(*arguments)
+    if !arguments[0].nil?
+      my_each { |index| return true unless arguments[0] == index }
+    elsif !block_given?
+      my_each { |index| return true unless index }
+    else
+      my_each { |index| return true unless yield(index) }
+    end
+    false
+  end
+
+  def my_any?(*arguments)
+    if !arguments[0].nil?
+      my_each { |index| return false unless arguments[0] == index }
+    elsif !block_given?
+      my_each { |index| return false unless index }
+    else
+      my_each { |index| return false unless yield(index) }
+    end
+    true
+  end
+
+  def my_none?(*args)
+    !my_any?(*args)
+  end
+
+  def my_count(arg = nil)
+    counter = 0
+    if arg
+      my_each { |element| counter += 1 if element == arg }
+    elsif !block_given?
+      counter = length
+    elsif !arg
+      my_each { |element| counter += 1 if yield element }
+    end
+    counter
+  end
+
+  def my_map
+    arr = []
+    my_each { |element| arr << yield(element) }
+    arr
+  end
+
+  def my_inject(number = nil, sym = nil)
+    if block_given?
+      accumulator = number
+      my_each { |index| accumulator = accumulator.nil? ? index : yield(accumulator, index) }
+      accumulator
+    elsif !number.nil? && (number.is_a?(Symbol) || number.is_a?(String))
+      accumulator = nil
+      my_each { |index| accumulator = accumulator.nil? ? index : accumulator.send(number, index) }
+      accumulator
+    elsif !sym.nil? && (sym.is_a?(Symbol) || sym.is_a?(String))
+      accumulator = number
+      my_each { |index| accumulator = accumulator.nil? ? index : accumulator.send(sym, index) }
+      accumulator
+    end
+  end
+end
+# rubocop:enable Metrics/CyclomaticComplexity
+# rubocop:enable Metrics/PerceivedComplexity
+def multiply_els(array)
+  puts '=====multiply_els====='
+  puts array.my_inject(1) { |value, i| value * i }
+end
```

Comment:
# multiply_els 
1. [ ] Accepts an array as an argument and multiplies all the elements of the array together using `#my_inject`
### Test 👉 `expect(actual).to eq 40`
|Expected|Got|
|:---:|:---:|
|`40`|`nil`|